Sources/PostgresNIO/New/Data/Optional+PSQLCodable.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  File.swift
+//  
+//
+//  Created by Fabian Fett on 13.01.21.
+//
+
+extension Optional: PSQLDecodable where Wrapped: PSQLDecodable {
+    static func decode(from byteBuffer: inout ByteBuffer, type: PSQLDataType, context: PSQLDecodingContext) throws -> Optional<Wrapped> {
+        preconditionFailure("This code path should never be hit.")
+        // The code path for decoding an optional should be:
+        //  -> PSQLData.decode(as: String?.self)
+        //       -> PSQLData.decodeIfPresent(String.self)
+        //            -> String.decode(from: type:)
+    }
+}
+
+extension Optional: PSQLEncodable where Wrapped: PSQLEncodable {
+    var psqlType: PSQLDataType {
+        switch self {
+        case .some(let value):
+            return value.psqlType
+        case .none:
+            return .null
+        }
```

Comment:
```suggestion
        self.map { $0.psqlType } ?? .null
```

Reply:
Do you mind if we keep the explicit way? While your proposed solution is shorter and more elegant. It always takes me minutes to parse this short form. And at least one lookup of `map` and `flatMap` overload in the context of `Optional`.

Reply:
Would you find `self?.psqlType ?? .null` easier to read? (It's certainly better than my first suggestion!)